### ISSUE TITLE
Regression: MPI_Win_start()/complete() with MPI_GROUP_EMPTY

### ISSUE THREAD

**Issue by dalcinl** [open-mpi/ompi#8677]
I don't consider myself an expert on OSC, but the following code used to work across MPI implementations. Now it is failing with current master.

Reproducer
```python
from mpi4py import MPI  # Implies MPI_Init_thread() with MPI_THREAD_MULTIPLE

buf = MPI.Alloc_mem(1024)
win = MPI.Win.Create(buf, 1, MPI.INFO_NULL, MPI.COMM_SELF)
win.Start(MPI.GROUP_EMPTY)
win.Complete()
win.Free()
MPI.Free_mem(buf)
```
Output:
```
Traceback (most recent call last):
  File "/tmp/rma.py", line 5, in <module>
    win.Complete()
  File "mpi4py/MPI/Win.pyx", line 592, in mpi4py.MPI.Win.Complete
mpi4py.MPI.Exception: MPI_ERR_RMA_SYNC: error executing rma sync
```

**Comment by jsquyres** [comment]
@hjelmn Does this code look correct?

**Comment by dalcinl** [comment]
Note that the following code runs to completion. If `Post()` with the empty group works, then I would argue that `Start()` with the empty group should also work. 

```python
from mpi4py import MPI
win = MPI.Win.Create(MPI.BOTTOM, 1, MPI.INFO_NULL, MPI.COMM_SELF)
win.Post(MPI.GROUP_EMPTY)
win.Wait()
win.Free()
```

**Comment by dalcinl** [comment]
@jsquyres I know the test looks a bit silly. Those are the silly things and corner cases I test for. I get what I deserve.

**Comment by dalcinl** [comment]
> @hjelmn Does this code look correct?

An MPI expert always needs to double check with a colleague about OSC semantics :rofl:.

**Comment by bosilca** [comment]
All these version look legit.

**Comment by hjelmn** [comment]
While odd that should work as I understand the standard.

**Comment by hjelmn** [comment]
When fixed we should add both a C (if it is missing) and fortran version of this to ompi-tests to cover.

**Comment by jsquyres** [comment]
> An MPI expert always needs to double check with a colleague about OSC semantics 🤣.

I stay far, far away from all MPI one-sided things.  😃

**Comment by hjelmn** [comment]
Why? One-sided is fast communication without the slowness that is MPI two-sided semantics 😁

**Comment by jsquyres** [comment]
This is how I think of MPI one-sided:

![explain-network](https://user-images.githubusercontent.com/617316/112052570-a601fc00-8b29-11eb-8e19-3d56faec5b48.gif)

**Comment by awlauria** [comment]
I addressed this in #8756 for osc/rdma.

The sync type wasn't being set, causing the SYNC error.
Also there was a couple issues freeing an empty group, the attached basic test runs clean for me now.

```
#include<mpi.h>
#include<stdio.h>
#include<stdlib.h>

int main() {
 //Start up MPI...
    MPI_Win win;
    char *buf;
    int rank;

    MPI_Init(0,0);
    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
    MPI_Group comm_group, group;

    if (rank == 0) {
        MPI_Win_create(buf,sizeof(int)*3,sizeof(int),
            MPI_INFO_NULL,MPI_COMM_WORLD,&win);
    }
    else {
        /* Rank 1 or 2 */
        MPI_Win_create(NULL,0,sizeof(int),
            MPI_INFO_NULL,MPI_COMM_WORLD,&win);
        MPI_Win_start(MPI_GROUP_EMPTY,0,win);
        /* Put into rank==0 according to my rank */
        /* Terminate the access epoch */
        MPI_Win_complete(win);

    }
    /* Free window and groups */
    MPI_Win_free(&win);
    return 0;
}
```
